activerecord/lib/active_record/associations.rb

```diff
@@ -324,6 +324,11 @@ def association(name) # :nodoc:
       association
     end
 
+    def load_async(*associations) # TODO: doc
```

Comment:
I'm fairly sure the use case here is to provide a shorthand that prevents people having to list all associations on a model if they want everything.

Given:

```rb
class MyProduct
  belongs_to :shop
  has_many :variants
  has_many :inventory_items
```

Rather than having to `product.load_async(:shop, :variants, :inventory_items)` it would be nice to have `product.load_async(:all_associations)`. Naming is just for illustration.

Reply:
I'm not convinced this is a common enough need to justify it, nor that it's a good idea. 

And if you really want to do that for some reason, it's quite easy to list all association names of a model (e.g. `Shop.reflections.keys`).

Reply:
Fair enough - I don't know whether it's common or not (yet), but in the particular site where I am planning to use this, we are are doing something along those lines - but, again, with the shorthand above if/when we need it, we have a solution so it's all good.